chore: small quality of life changes for TweakDbId and ItemId

Roms1383 opened the pull request:
Feel free to dismiss this PR if you think it's too much, but it basically:

- promotes both types as `Copy` since `TweakDbId` is 8 bytes only and `ItemId` 16 bytes
- makes `ItemId` `new_from` method `const` and `inline`
- adds `const` `get_tdbid` and `is_of_tdbid` convenience methods (as member functions, opposed to cyberdoc which describes them as `static func`)
- adds a small test to ensure conversions

Roms1383 commented:
▶️ PR's need to hold back for a bit: ongoing work is still going on in RED4ext.SDK.

jac3km4 commented on red4ext-sys/src/interop.rs:
```diff
@@ -220,10 +234,12 @@ pub enum GameEItemIdFlag {
 #[derive(Debug, Clone, Copy, PartialEq, Eq)]
 pub struct Seed(u32);
 
+const DEFAULT_ITEMID_RNG_SEED: u32 = 2;
```
```suggestion
const DEFAULT_ITEM_ID_RNG_SEED: u32 = 2;
```

Roms1383 replied:
addressed in 06b0632

Roms1383 commented:
Removing `example` from workspace's members break `just dev` command, currently addressing that.

jac3km4 commented:
I removed it from the workspace because you can't have two toolchain configs in a single workspace. It'll need to remain separate or be moved to a different repo maybe.

Roms1383 commented:
> I removed it from the workspace because you can't have two toolchain configs in a single workspace. It'll need to remain separate or be moved to a different repo maybe.

I fixed it by defaulting to `nightly` with `rust-toolchain.toml`, then defined the `example` appropriately in `red4ext` package and updating the related `justfile` commands.

It builds fine on my machine, currently testing the lib in-game, I'll provide feedback in a couple of minutes 👌

jac3km4 commented:
I think the workspace should build against stable Rust, not nightly, because the main functionality does not require any nightly features (and IMO it shouldn't). I think it makes sense to separate things and build the workspace on stable and some examples on nightly, if they require that.

Roms1383 commented:
It works like a charm ! ❤️ 🦀 

There's this minor issue with the `global operator`, but it has always been here as far as I know so no regression:
```sh
> LogChannel(CName.new("DEBUG"), OperatorAdd(2000, 77))
2000BerserkChargesDecayStartDelay
```

jac3km4 commented:
What is the issue here? If you're invoking the operator from CET, I think it's probably just calling the operator for the wrong type, that doesn't have anything to do with this library as far as I can see

Roms1383 commented:
> What is the issue here? If you're invoking the operator from CET, I think it's probably just calling the operator for the wrong type, that doesn't have anything to do with this library as far as I can see

Yes that's why I really don't mind it 👌 Just stating.

jac3km4 commented on .github/workflows/ci.yml:
```diff
@@ -45,7 +45,9 @@ jobs:
           submodules: true
       - run: echo "LIBCLANG_PATH=$((gcm clang).source -replace "clang.exe")" >> $env:GITHUB_ENV
-      - uses: dtolnay/rust-toolchain@stable
+      - uses: dtolnay/rust-toolchain@nightly
```
As I said above, I don't think we should build the workspace on nightly. Everything except for the examples should be built with stable rust.

Roms1383 replied:
Ok let me process with the changes, I'll try to be quick.

Roms1383 replied:
Sorry I didn't understand correctly at first. Let's see the pipeline, it should be fine by now.

Roms1383 replied:
@jac3km4 should we have CI test run also on nightly ? or change the test command to not include `--all-features` (which is the reason why `cargo test` is also run for `example`) ?

jac3km4 commented on justfile:
```diff
@@ -40,7 +41,9 @@ install target='release':
  }
- @if ('{{target}}' -EQ 'debug') { cargo +nightly build --example example --features="macros"; } else { cargo +nightly build --example example --features="macros" --release; }
+ cd '{{example_dir}}'; \
+ if ('{{target}}' -EQ 'debug') { cargo +nightly build; } else { cargo +nightly build --release; } \
```
the final artifact will now be under a different path:
`red4ext-rs\example\target\release\example.dll`
I think you'll need to update the copy paths too

Roms1383 replied:
you're completely right, I tried to rush and thought I addressed it while I clearly forgot.
I fixed it in cbd8a14, also simplified the logic to avoid having to `cd ..` here and there.
I also applied the same change in subsequent 1266802 to the CI `build` job, to make sure `example` gets compiled properly.

jac3km4 commented:
lgtm!
